src/rabbit_lager.erl

```diff
@@ -182,9 +182,15 @@ configure_lager() ->
     end,
     case application:get_env(lager, log_root) of
         undefined ->
-            application:set_env(lager, log_root,
-                                application:get_env(rabbit, lager_log_root,
-                                                    undefined));
+            %% Setting env var to 'undefined' is different from not
+            %% setting it at all, and lager is sensitive to this
+            %% difference.
+            case application:get_env(rabbit, lager_log_root, undefined) of
+                {ok, FromRabbitConfig} ->
```

Comment:
This line will never match. `application:get_env/3` returns value itself or a default value.
Did you wanted to use `application:get_env(rabbit, lager_log_root)` instead?